src/Database/Connection.php

```diff
@@ -727,6 +728,39 @@ protected function _newLogger(StatementInterface $statement)
     }
 
     /**
+     * Checks if the driver supports EXPLAIN statement for the specified query.
+     *
+     * @param string $query The query to be explained
+     * @return bool
+     */
+    public function canExplain($query)
+    {
+        $query = (string)$query;
+
+        return preg_match('/^\s*(SELECT|INSERT|UPDATE|DELETE)\s/i', $query)
+            && $this->_driver->explainSQL($query) !== false;
+    }
+
+    /**
+     * Executes an EXPLAIN statement for the specified query.
+     *
+     * @param string $query The query to be explained
+     * @param array $params list or associative array of params to be interpolated in $query as values
+     * @param array $types list or associative array of types to be used for casting values in query
+     * @return \Cake\Database\StatementInterface
+     */
+    public function explain($query, array $params = [], array $types = [])
```

Comment:
Parsing SQL for a feature like this is not a good idea in my opinion. It will add a large amount of complexity and not an equal amount of benefit. Could `explain()` work on `Query` objects *or* strings?